impeller/playground/backend/metal/playground_impl_mtl.mm

```diff
@@ -76,6 +76,8 @@
   auto context = ContextMTL::Create(
       ShaderLibraryMappingsForPlayground(), is_gpu_disabled_sync_switch_,
       "Playground Library",
+      // TODO(tbd): Switch to BGRA10_XR to match iOS's behavior when the ci bots
+      //            support it.
```

Comment:
This unfortunately means we can't rip out the advanced blend without ripping out the tests.

Reply:
We could rewrite the test to be a framework golden test.  I'm not sure where we stand with framework golden tests that run with impeller, maybe @jonahwilliams knows.

Reply:
The bot that failed this test was intel.  I suspect BGRA10_XR support is linked to having an Intel or ARM mac.

Reply:
we could pin or ifdef

Reply:
Would the test just randomly sometime not run then?  I think what kind of bot you get is luck of the draw?

Reply:
The [Metal Feature Set Tables](https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf) show that this format is only available on `Apple3` family and above and never on the Intel family. I find it annoying that there is no Metal API to just query support for a specific format. `Apple2` which doesn't support this format still covers iPhone 6 which I think we do support per the minimum version guarantees.

Perhaps we could have a Metal specific feature format availability check (that internally uses the family checks to determine availablility) and use that at runtime. An ifdef would still leave us in trouble on the iPhone 6.

Reply:
I think the way to check is to just try to make a texture with that format.  We can do that if we want.  The problem for the test though is I want to make sure it gets run.

Reply:
We are already using BGR10_XR too for over a year and haven't gotten bug reports.  It's only used on wide gamut content that doesn't have alpha, so maybe it slipped through if it's only a problem on a really old phone.

Reply:
> Are there some gates to wide gamut maybe?

Nope, just on by default.

Reply:
wait, no there is a check for support in flutterview.mm:

`return self.screen.traitCollection.displayGamut != UIDisplayGamutSRGB`

Reply:
There are gates to wide gamut, not all of our device types support it: https://github.com/flutter/engine/blob/main/shell/platform/darwin/ios/framework/Source/FlutterView.mm#L58-L69

Reply:
Okay, so we are good on using BGRA10_XR, but that doesn't answer the original question: what can we do about testing?  We have a selection of CI bots that can't run our tests.  Options include:
1) only run the tests on bots that support it
1) finding some way to guarantee the test is run on a device that supports it
1) rewrite the test to be a framework test that runs on device lab with impeller

Reply:
There is already a devicelab test that asserts wide gamut works. We don't have a mechanism to run impeller goldens in the framework that isn't on device or using swiftshader. Add it to the list of standing complaints and lets move on.

Reply:
Ah, that makes sense. Presumably, the devices that don't support these formats also don't have displays capable of that format.

Reply:
I don't think we need specific wide gamut goldens in the engine if the alpha behavior is the same. We certainly shouldn't keep broken behavior around because its easier to test.

Reply:
> We certainly shouldn't keep broken behavior around because its easier to test.

Agreed, that isn't one of the options I proposed.  Ideally we test this in the engine, I'm trying to understand if there is a way for us to do that.  Otherwise, I'll have to rewrite the test at the device lab somehow and we can delete it in the engine.  Writing it in the devicelab will mean some sort of pidgin golden test.  I can probably come up with something.

Reply:
I don't think that will carry its weight as a devicelab test. I would restrict the wide gamut tests to supported devices and ask @zanderso for help wiring it up as a separate test shard on CI.

Reply:
Zach pointed to this example on discord: https://github.com/flutter/engine/blob/main/ci/builders/mac_unopt.json#L125

It seems like we already may be running on an arm64 mac, but we are building the x86 variant and executing it with rosetta:
https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20mac_host_engine/29358/overview

Reply:
That mac_unopt build looks like this on CI https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20Production%20Engine%20Drone/310981/overview. I think it's building arm64 artifacts and running natively on the arm64 machine in CI. It's run_tests.py invocation looks like this: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8751008061587291905/+/u/test:_Host_Tests_for_host_debug_unopt_arm64/execution_details

Reply:
Yea, i just moved it over to the arm64 build.  That seems to work so hopefully adding it there is fine.  I was just mentioning that since it seems that maybe the x86 requirement isn't respected but the arm64 one is?  It's unrelated to what I need to implement this.

Reply:
Okay, I finally got it sorted an moved the golden tests over to arm64.  Luckily there doesn't seem to be a golden diff.  Thanks everyone.